src/ast/code_generator.ml

```diff
@@ -0,0 +1,585 @@
+(* SPDX-License-Identifier: AGPL-3.0-or-later *)
+(* Copyright © 2021-2024 OCamlPro *)
+(* Written by the Owi programmers *)
+
+open Types
+open Binary
+open Spec
+open Syntax
+
+let type_env (m : modul) (func_ty : binary param_type * binary result_type)
+  (owi_funcs : (string * int) list) =
+  object
```

Comment:
Why did you decide to go with an object ?

Reply:
I need to pass in the types of global variables, function parameters and results, symbolic execution functions and quantified variables to generate correct code, and I don't want to complicate the type. So I think maybe an object helps me to abstract away all these?

Reply:
Would you like to try with modules instead of objects ? There is currently no objects in the codebase (there's object *types* only) and I would like to avoid them if possible. :sweat_smile:

Reply:
I also expect a record to be enough there.